File: frontend/src/types/inventory.ts
export interface Inventory {
  id: number;
  branchId: number;
  productId: number;
  stock: number;
  minStockThreshold: number;
  version: number;
}

export interface InventoryRequest {
  branchId: number;
  productId: number;
  stock: number;
  minStockThreshold: number;
}

File: backend/src/main/java/com/optiplant/inventory/domain/entity/Inventory.java
package com.optiplant.inventory.domain.entity;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Getter @Setter @NoArgsConstructor @AllArgsConstructor @Builder
@Table(uniqueConstraints = @UniqueConstraint(columnNames = {"branch_id", "product_id"}))
public class Inventory {
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "branch_id", nullable = false)
    private Branch branch;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "product_id", nullable = false)
    private Product product;
    
    @Column(nullable = false)
    private Integer stock;
    
    @Column(nullable = false)
    private Integer minStockThreshold;
    
    @Version
    private Integer version;
}
